src/reconstruction.jl

```diff
@@ -0,0 +1,214 @@
+using Statistics
+using LinearAlgebra
+export AbstractSpatialEmbedding
+export SpatioTemporalEmbedding, STE
+export outdim
+export AbstractBoundaryCondition, PeriodicBoundary, ConstantBoundary
+
+import DynamicalSystemsBase: reconstruct
+
+#####################################################################################
+#                 Spatio Temporal Delay Embedding Reconstruction                    #
+#####################################################################################
+"""
+    AbstractSpatialEmbedding <: AbstractEmbedding
+Super-type of spatiotemporal embedding methods.
+Use `subtypes(AbstractSpatialEmbedding)` for available methods.
+"""
+abstract type AbstractSpatialEmbedding{T,Φ,BC,X} <: AbstractEmbedding end
+const ASE = AbstractSpatialEmbedding
+"""
+    AbstractBoundaryCondition
+Super-type of boundary conditions for [`SpatioTemporalEmbedding`](@ref).
+Use `subtypes(AbstractBoundaryCondition)` for available methods.
+"""
+abstract type AbstractBoundaryCondition end
+
+"""
+	ConstantBoundary(C)
+Constant boundary condition type. Enforces constant boundary conditions
+when passed to [`SpatioTemporalEmbedding`](@ref)
+by filling missing out-of-bounds values in the reconstruction with
+parameter `C`.
+"""
+struct ConstantBoundary{T} <: AbstractBoundaryCondition
+    C::T
```

Comment:
Don't capitalize this `C`. (`c` instead).

I will be pulling your hair with these ridiculous comments I make, I am sure.